Automattic/wp-calypso
# Checkout: Remove "onInitialize" arguments from useShoppingCartManager, take 2

#### Changes proposed in this Pull Request

This is a second attempt on https://github.com/Automattic/wp-calypso/pull/45928 which was reverted in https://github.com/Automattic/wp-calypso/pull/46158 because [this commit](https://github.com/Automattic/wp-calypso/pull/45928/commits/1ec1417687ff816e6a02dedf947be72aadb648e3) was causing contact form validation to never complete.

This PR is the same as the original, but with the offending commit removed.

#### Original PR testing instructions

- Visit calypso, enable record-analytics debugging by typing `localStorage.setItem('debug', 'calypso:composite-checkout:record-analytics')` into your console, and then reloading the page.
- Visit checkout with a product in the URL like `/checkout/example.com/premium`.
- Verify that checkout loads without errors and that you see the product added to your cart.
- When loading checkout, make sure that the `CART_INIT_COMPLETE` event happens in the console, that it happens after the cart is initialized, and that it only happens once.
- When loading checkout, make sure that the `CART_ADD_ITEM` event happens in your console, that it has the correct product listed, and that it happens only once.
- Remove all products from your cart.
- Verify that you are redirected to the `/plans` page.
- Visit checkout with a product and a coupon in the URL like `/checkout/example.com/premium?coupon=FOOBAR`.
- Verify that checkout loads without errors and that you see the product and the coupon added to your cart.

##### New testing steps

- Add a domain to your cart and visit checkout.
- Verify that checkout finishes loading.
